### DISCUSSION TITLE
Sharing Environments Between Repositories?

### DISCUSSION THREAD

**Discussion by Porter-dm** [community/community#37978] [Actions]
I have over a hundred repositories and dozens of environments to which I want to deploy them.  

Instead of creating each environment in every single repository, is it possible to define an environment once and use it across multiple repositories?

**Comment by Porter-dm** [comment]
If not, is there any other workaround to creating thousands of different environments?

Right now, each environment only consists of three variables, which are stored as environment secrets (the name of the environment, the type of environment [such as prod or dev] , and the AWS account number).  Perhaps there's a better way of packaging and selecting those three values other than repository environments?

**Comment by mickeygousset** [comment]
Currently environments are repository centric, unfortunately, so you would have to define them in multiple repos.

How much overlap is there in those values?  Meaning, for example, is the AWS account number used multiple times?  You might be able to come up with a naming convention and use organizational secrets for the values. Then you can assign those secrets only to repos that need them.

**Reply by ssbarnea** [reply]
I have the same kind of issue. Even if I already have some reusable github actions, they need some credentials, some which are not expected to be configurable by the callers (consuming repositories). 

Example notification action which is expected to send notifications by email or different chat IM. They need cloud credentials and having to configure these on each repository that would want to make use of these reusable action does not make any sense. In fact it would increase the security risk due to having them store in far more places.

We need a way to allow a reusable action to use secrets from its own repository.

Current situation is a maintenance nightmare, one that effectively prevents reuse of huge number of actions as that does not really scale.

To make the situation even more interesting, our company has something like 10 github repositories and we do often reuse action across these organizations. This means that a reusable action should declare itself which consumers it trusts as we don't want someone else to start making use of its secrets.

I suspect something like
```
trust-orgs:
- acme
- acme-contribs
```

**Comment by blagar2** [comment]
Hi ,
For info, there is a discussion about it here: : https://github.com/orgs/community/discussions/15379
 I hope that this feature will be implemented.
